tests/Feature/EnrollmentControllerTest.php

```diff
@@ -15,40 +16,84 @@ class EnrollmentControllerTest extends TestCase
 
     public string $uri = '/api/v1/enrollments';
 
-    public function testEnrollmentsAreCreated()
+    public $seeder = TestSeeder::class;
+
+    // Each controllers test should store these properties for assertion
+    public $single_json_structure = [
+        'data' => [
+            'user_id',
+            'course_id',
+            'provider_platform_id',
+            'enrollment_state',
+            'links',
+            'provider_enrollment_id',
+            'provider_start_at',
+            'provider_end_at',
+            'created_at',
+            'updated_at',
+        ],
+    ];
+
+    public $array_json_structure = [
+        'data' => [
+            '*' => [
+                'user_id',
+                'course_id',
+                'provider_platform_id',
+                'enrollment_state',
+                'links',
+                'provider_enrollment_id',
+                'provider_start_at',
+                'provider_end_at',
+                'created_at',
+                'updated_at',
+            ],
+        ],
+    ];
+
+    public function testAdminCanCreateAndAccessEnrollments()
     {
-        $user = \App\Models\User::factory()->admin()->createOne();
-        Enrollment::factory(10)->create();
+        $this->seed($this->seeder);
+        $user = User::factory()->createOne();
 
-        $response = $this->actingAs($user)->get($this->uri, [
-            'Accept' => 'application/json',
-        ]);
+        $admin = \App\Models\User::factory()->admin()->createOne();
 
-        $response->assertOk();
-        $this->assertCount(10, $response['data']);
+        $users_enrollments = Enrollment::factory()->count(5)->forUser($user->id)->make();
+
+        foreach ($users_enrollments as $enrollment) {
+            $fail = $this->actingAs($user)->postJson($this->uri, $enrollment->toArray());
+            $fail->assertStatus(403);
+            $response = $this->actingAs($admin)->postJson($this->uri, $enrollment->toArray());
+            $response->assertStatus(201);
+            $response->assertJsonStructure($this->single_json_structure);
+        }
+        $responseUser = $this->actingAs($user)->getJson($this->uri);
+
+        $responseAdmin = $this->actingAs($admin)->getJson($this->uri);
+
+        $responseAdmin->assertJsonCount(10, 'data');
+
+        $responseUser->assertJsonCount(5, 'data');
     }
 
-    // .... i have no words for some of these tests. We can do better
```

Comment:
"We can do better" ... is that a specifically or a general philosophical statement haha?

Reply:
Lol I guess technically both? I forgot I committed that :rofl: